highgui Qt backend broken in master

<!--
If you have a question rather than reporting a bug please go to https://forum.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)
<!-- Example
- OpenCV => 4.2
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2017
-->

- OpenCV => master https://github.com/opencv/opencv/tree/13c6eb42e9788172c10132a35badacdb23de034b
- Operating System / Platform => openSUSE Tumbleweed
- Compiler => gcc 11.2



##### Steps to reproduce

Build OpenCV with Qt backend.

```cmake
cmake .. -DBUILD_LIST=highgui -DWITH_QT=ON -DWITH_GTK=OFF -DHIGHGUI_ENABLE_PLUGINS=OFF
```

Run minimal program:
```c++
#include <opencv2/highgui.hpp>

int main()
{
    cv::namedWindow("img");
}
```

Result in
```
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.5.4-pre) /home/huangqinjin/Projects/opencv/modules/highgui/src/window.cpp:1257: error: (-2:Unspecified error) The function is not implemented. Rebuild the library with Windows, GTK+ 2.x or Cocoa support. If you are on Ubuntu or Debian, install libgtk2.0-dev and pkg-config, then re-run cmake or configure script in function 'cvNamedWindow'
```


##### Issue submission checklist

 - [x] I report the issue, it's not a question
   <!--
   OpenCV team works with forum.opencv.org, Stack Overflow and other communities
   to discuss problems. Tickets with question without real issue statement will be
   closed.
   -->
 - [x] I checked the problem with documentation, FAQ, open issues,
       forum.opencv.org, Stack Overflow, etc and have not found solution
   <!--
   Places to check:
   * OpenCV documentation: https://docs.opencv.org
   * FAQ page: https://github.com/opencv/opencv/wiki/FAQ
   * OpenCV forum: https://forum.opencv.org
   * OpenCV issue tracker: https://github.com/opencv/opencv/issues?q=is%3Aissue
   * Stack Overflow branch: https://stackoverflow.com/questions/tagged/opencv
   -->
 - [x] I updated to latest OpenCV version and the issue is still there
   <!--
   master branch for OpenCV 4.x and 3.4 branch for OpenCV 3.x releases.
   OpenCV team supports only latest release for each branch.
   The ticket is closed, if the problem is not reproduced with modern version.
   -->
 - [x] There is reproducer code and related data files: videos, images, onnx, etc
   <!--
   The best reproducer -- test case for OpenCV that we can add to the library.
   Recommendations for media files and binary files:
   * Try to reproduce the issue with images and videos in opencv_extra repository
     to reduce attachment size
   * Use PNG for images, if you report some CV related bug, but not image reader
     issue
   * Attach the image as archive to the ticket, if you report some reader issue.
     Image hosting services compress images and it breaks the repro code.
   * Provide ONNX file for some public model or ONNX file with with random weights,
     if you report ONNX parsing or handling issue. Architecture details diagram
     from netron tool can be very useful too. See https://lutzroeder.github.io/netron/
   -->

--- a/modules/highgui/CMakeLists.txt
+++ b/modules/highgui/CMakeLists.txt
@@ -51,6 +51,8 @@ set(OPENCV_HIGHGUI_BUILTIN_BACKEND "")
 
 if(HAVE_QT)
   set(OPENCV_HIGHGUI_BUILTIN_BACKEND "QT${QT_VERSION_MAJOR}")
+  add_definitions(-DHAVE_QT)
+
   if(QT_VERSION_MAJOR GREATER 4)
     # "Automoc" doesn't work properly with opencv_world build, use QT<ver>_WRAP_CPP() directly
     #set(CMAKE_AUTOMOC ON)
@@ -75,6 +77,7 @@ if(HAVE_QT)
 
     set(qt_deps Core Gui Widgets Test Concurrent)
     if(HAVE_QT_OPENGL)
+      add_definitions(-DHAVE_QT_OPENGL)
       list(APPEND qt_deps OpenGL)
     endif()
 

